src/entry.rs

```diff
@@ -574,40 +575,6 @@ impl<R: Read + Unpin> EntryFields<R> {
                 )));
             }
 
-            if let Ok(metadata) = fs::metadata(dst).await {
-                let result = if metadata.is_symlink() || !metadata.is_dir() {
-                    remove_file(dst).await
-                } else {
-                    remove_dir_all(dst).await
-                };
-
-                result.map_err(|err| {
-                    Error::new(
-                        err.kind(),
-                        format!(
-                            "{} when removing existing entity for linking {} to {}",
-                            err,
-                            src.display(),
-                            dst.display()
-                        ),
-                    )
-                })?;
-            }
-
-            if fs::symlink_metadata(dst).await.is_ok() {
-                remove_file(dst).await.map_err(|err| {
-                    Error::new(
-                        err.kind(),
-                        format!(
-                            "{} when removing existing symlink for linking {} to {}",
-                            err,
-                            src.display(),
-                            dst.display()
-                        ),
-                    )
-                })?;
-            }
```

Comment:
Some of this was here in https://github.com/astral-sh/tokio-tar/commit/c6921b7c0e9122d38aa73a357c441fde08b49ad9, but it seems preferable to mirror upstream?